### PR TITLE
Fix Travis configuration for releases #22

### LINKED ISSUE #22: Travis errors on release
Travis-CI produces an error on a release build plus runs release script for each Go version.

### PR DESCRIPTION
**Pull request by AlfredBroda**
Splits build and release stages into separate stages. Fixes #22

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,15 +2,18 @@ language: go
...
-    # Build the binary
-    - make release
-  skip_cleanup: true
-  file: target/executor-linux-amd64.zip
-  on:
-    tags: true
+
+jobs:
+  include:
+    - script: make
```
Comment (medzin) [inline .travis.yml]: Make required changes in Makefile not here (adding lint as coveralls dependency).

Comment (AlfredBroda) [reply .travis.yml]: Isn't this more transparent? I was comparing to this kind of configuration: https://github.com/allegro/marathon-consul/blob/master/.travis.yml#L12

Comment (medzin) [reply .travis.yml]: Maybe separate Makefile target that combines build and coveralls for Travis will be better?

Comment (janisz) [reply .travis.yml]: In marathon-consul we always build package as a CI step just to be sure test are passing and application is buildable.

File: metrics/system.go [hunk 2]
```diff
@@ -32,7 +32,11 @@ func CPUTime() (float64, error) {
 // goroutine.
 func CaptureCPUTime(interval time.Duration) {
 	cpuUtilizationGauge := metrics.NewGaugeFloat64()
-	metrics.Register("runtime.CpuStats.Utilization", cpuUtilizationGauge)
+	err := metrics.Register("runtime.CpuStats.Utilization", cpuUtilizationGauge)
```
Comment (janisz) [inline metrics/system.go]: How is this change related to travis?

Comment (AlfredBroda) [reply metrics/system.go]: That is what was making the build fail when linter was ran.

File: Makefile [hunk 3]
```diff
@@ -69,3 +69,5 @@ test-deps:
 		(go get github.com/modocache/gover)
 	@which goveralls > /dev/null || \
 		(go get github.com/mattn/goveralls)
+
+travis-ci: lint test build coveralls
```
Comment (medzin) [inline Makefile]: test here is redundant

File: .travis.yml [hunk 4]
```diff
@@ -2,15 +2,17 @@ language: go
...
-    # Build the binary
-    - make release
-  skip_cleanup: true
-  file: target/executor-linux-amd64.zip
-  on:
-    tags: true
+
+jobs:
+  include:
+    - script: make travis-ci
```
Comment (janisz) [inline .travis.yml]: I prefer just `make release travis-ci` with no stages.

Comment (AlfredBroda) [reply .travis.yml]: Stage allows us to limit building binary packages to only build on tags and to have a single release stage regardless of how many version specific test builds we do (Go 1.9.x, master etc).
https://docs.travis-ci.com/user/build-stages/deploy-github-releases/
https://docs.travis-ci.com/user/conditional-builds-stages-jobs/#Conditional-Stages
